server/authz/policy.rego

```diff
@@ -430,9 +430,10 @@ allow {
 # Team admin and maintainers can read and write policies for their teams
 allow {
   not is_null(object.team_id)
-  object.team_id == subject.teams[_].id
+  team_id := subject.teams[_].id
+  object.team_id == team_id
   object.type == "policy"
-  team_role(subject, subject.teams[_].id) == [admin,maintainer][_]
```

Comment:
This was the bug, if a user was an observer of `Team1` and an admin of `Team2`, they could "write" policies of `Team1`.